src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -96,16 +97,34 @@ private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, A
 
             if (activity.IsAllDataRequested)
             {
-                activity.SetTag(SemanticConventions.AttributeHttpMethod, request.Method);
-                activity.SetTag(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host);
-                if (!request.RequestUri.IsDefaultPort)
+                // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md
+                if (Options.HttpSemanticConvention.HasFlag(HttpSemanticConvention.Old))
```

Comment:
These could be extracted into boolean variables to avoid checking them repeatedly. This can be done in another PR.

Reply:
The pattern used in the other libraries won't work here because this is a static class.

Reply:
Interesting. I'm not sure why this class uses a static instance of the options.

However, you should still be able to extract them into `static bool` variables in this case and use them instead of checking if the `enum` has a flag repeatedly.

Reply:
Not just a static instance of the options, the entire class is static.

I changed the Options field to a property so I can use the setter to evaluate the booleans in this commit: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4639/commits/b91509ef80a791c45ed1cd0a88e6c8f39a2254ca